src/terminal/input/terminalInput.cpp

```diff
@@ -440,120 +440,150 @@ static bool _translateDefaultMapping(const KeyEvent& keyEvent,
-                 keyEvent.GetVirtualKeyCode() == VK_SPACE))
-            {
-                // For Alt+Ctrl+Key messages, the UnicodeChar is NOT the Ctrl+key char, it's null.
-                //      So we need to get the char from the vKey.
-                //      EXCEPT for Alt+Ctrl+Space. Then the UnicodeChar is space, not NUL.
-                auto wchPressedChar = gsl::narrow_cast<wchar_t>(MapVirtualKeyW(keyEvent.GetVirtualKeyCode(), MAPVK_VK_TO_CHAR));
-                // This is a trick - C-Spc is supposed to send NUL. So quick change space -> @ (0x40)
-                wchPressedChar = (wchPressedChar == UNICODE_SPACE) ? 0x40 : wchPressedChar;
-                if (wchPressedChar >= 0x40 && wchPressedChar < 0x7F)
-                {
-                    //shift the char to the ctrl range
-                    wchPressedChar -= 0x40;
-                    _SendEscapedInputSequence(wchPressedChar);
-                    keyHandled = true;
-                }
-            }
+    const auto senderFunc = [this](const std::wstring_view seq) noexcept
+    {
+        _SendInputSequence(seq);
+    };
 
-            // If a modifier key was pressed, then we need to try and send the modified sequence.
-            if (!keyHandled && keyEvent.IsModifierPressed())
-            {
-                // Translate the key using the modifier table
-                keyHandled = _searchWithModifier(keyEvent, senderFunc);
-            }
-            // ALT is a sequence of ESC + KEY.
-            if (!keyHandled && keyEvent.GetCharData() != 0 && keyEvent.IsAltPressed())
-            {
-                _SendEscapedInputSequence(keyEvent.GetCharData());
-                keyHandled = true;
-            }
-            if (!keyHandled && keyEvent.IsCtrlPressed())
-            {
-                if ((keyEvent.GetCharData() == UNICODE_SPACE) || // Ctrl+Space
-                    // when Ctrl+@ comes through, the unicodechar
-                    // will be '\x0' (UNICODE_NULL), and the vkey will be
-                    // VkKeyScanW(0), the vkey for null
-                    (keyEvent.GetCharData() == UNICODE_NULL && keyEvent.GetVirtualKeyCode() == LOBYTE(VkKeyScanW(0))))
-                {
-                    _SendNullInputSequence(keyEvent.GetActiveModifierKeys());
-                    keyHandled = true;
-                }
-            }
+    // If a modifier key was pressed, then we need to try and send the modified sequence.
+    if (keyEvent.IsModifierPressed() && _searchWithModifier(keyEvent, senderFunc))
+    {
+        return true;
+    }
 
-            if (!keyHandled)
-            {
-                // For perf optimization, filter out any typically printable Virtual Keys (e.g. A-Z)
-                // This is in lieu of an O(1) sparse table or other such less-maintainable methods.
-                // VK_CANCEL is an exception and we want to send the associated uChar as is.

```

Comment:
Is this a worthwhile optimization to keep? I'd be interested in seeing the real-world performance impact.

Reply:
`_translateDefaultMapping` is an extremely simple and fast method (<1us I‘m sure) and I suppose it‘s not necessary to retain such an optimization.
On the other hand the intention of the previous code is easy to understand and doesn’t add much complexity.
I‘ll gladly add it back in. 👍

(Our of curiosity I‘ll make sure to post the performance impact of that function tomorrow. 🙂)

Reply:
I believe you that it's incredibly minor, so I am okay leaving out the optimization. It used to be a std::map, but then it was replaced with an std::array and not revisited. No need to share performance traces as a requirement for landing this :smile: